src/H5O.c

```diff
@@ -1365,29 +1373,20 @@ H5Ovisit_by_name3(hid_t loc_id, const char *obj_name, H5_index_t idx_type, H5_it
 } /* end H5Ovisit_by_name3() */
 
 /*-------------------------------------------------------------------------
- * Function:    H5Oclose
- *
- * Purpose:     Close an open file object.
+ * Function:    H5O__close_api_common
  *
- *              This is the companion to H5Oopen. It is used to close any
- *              open object in an HDF5 file (but not IDs are that not file
- *              objects, such as property lists and dataspaces). It has
- *              the same effect as calling H5Gclose, H5Dclose, or H5Tclose.
+ * Purpose:     This is the common function for closing an object.
  *
  * Return:      SUCCEED/FAIL
  *
- * Programmer:	James Laird
- *		July 14 2006
- *
  *-------------------------------------------------------------------------
  */
-herr_t
-H5Oclose(hid_t object_id)
+static herr_t
+H5O__close_api_common(hid_t object_id)
```

Comment:
Definitely the right idea, but maybe another name would be better since it's pretty different from the other api_common routines?  How about H5O__close_check and have it return an htri_t for yes / no / fail?   Then the H5Oclose / H5Oclose_async calls could call it and fail when they receive a no or fail return value.